app/tasks.py

```
@@ -0,0 +1,279 @@
+    """
+
+    try:
+        llm = ChatOpenAI(model_name=query_generation_model, temperature=temperature)
+        prompt = PromptTemplate.from_template(query_generation_prompt)
+        chain = prompt | llm | StrOutputParser()
+        raw_response = chain.invoke({"rfp_content": rfp_text})
+
+        json_str = extract_json_from_text(raw_response)
+        result = json.loads(json_str)
+        return result
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)